resources/BUILD.gn

```diff
@@ -0,0 +1,163 @@
+    deps += [ "//ui/views/resources" ]
+  }
+}
+
+repack("xwalk_resources_100_percent_pak") {
+  output = "$root_out_dir/xwalk_100_percent.pak"
+  sources = [
+    "$root_gen_dir/blink/public/resources/blink_image_resources_100_percent.pak",
+    "$root_gen_dir/components/components_resources_100_percent.pak",
+    "$root_gen_dir/content/app/resources/content_resources_100_percent.pak",
+    "$root_gen_dir/ui/resources/ui_resources_100_percent.pak",
+  ]
+  deps = [
+    "//components/resources",
+    "//content/app/resources:resources_grit",
+    "//third_party/WebKit/public:image_resources_grit",
+    "//ui/resources",
+  ]
+  if (toolkit_views) {
+    deps += [ "//ui/views/resources:resources_grd_grit" ]
+    sources +=
+        [ "$root_gen_dir/ui/views/resources/views_resources_100_percent.pak" ]
+  }
+}
+
+repack("xwalk_resources_200_percent_pak") {
+  output = "$root_out_dir/xwalk_200_percent.pak"
+  sources = [
+    "$root_gen_dir/blink/public/resources/blink_image_resources_200_percent.pak",
+    "$root_gen_dir/components/components_resources_200_percent.pak",
+    "$root_gen_dir/content/app/resources/content_resources_200_percent.pak",
+    "$root_gen_dir/ui/resources/ui_resources_200_percent.pak",
+  ]
+  deps = [
+    "//components/resources",
+    "//content/app/resources:resources_grit",
+    "//third_party/WebKit/public:image_resources_grit",
+    "//ui/resources",
+  ]
+  if (toolkit_views) {
+    sources +=
+        [ "$root_gen_dir/ui/views/resources/views_resources_200_percent.pak" ]
+    deps += [ "//ui/views/resources:resources_grd_grit" ]
+  }
+}
+
+repack("xwalk_resources_300_percent_pak") {
+  output = "$root_out_dir/xwalk_300_percent.pak"
+  sources = [
+    "$root_gen_dir/components/components_resources_300_percent.pak",
+    "$root_gen_dir/ui/resources/ui_resources_300_percent.pak",
+  ]
+  deps = [
+    "//components/resources",
+    "//ui/resources",
+  ]
+  if (toolkit_views) {
+    sources +=
+        [ "$root_gen_dir/ui/views/resources/views_resources_300_percent.pak" ]
+    deps += [ "//ui/views/resources:resources_grd_grit" ]
+  }
+}
+
+grit("generate_xwalk_resources") {
+  outputs = [
+    "$root_gen_dir/xwalk/resources/grit/xwalk_resources.h",
+    "$root_gen_dir/xwalk/resources/grit/xwalk_resources_map.cc",
+    "$root_gen_dir/xwalk/resources/grit/xwalk_resources_map.h",
+    "$root_gen_dir/xwalk/resources/xwalk_resources.pak",
+    "$root_gen_dir/xwalk/resources/xwalk_resources.rc",
+  ]
+  source = "//xwalk/runtime/resources/xwalk_resources.grd"
+  resource_ids = "//xwalk/resources/resource_ids"
+}
+
+copy("xwalk_resources") {
+  outputs = [
+    "$root_gen_dir/xwalk/grit/{{source_file_part}}",
+  ]
+  sources = [
+    "$root_gen_dir/xwalk/resources/grit/xwalk_resources.h",
+    "$root_gen_dir/xwalk/resources/grit/xwalk_resources_map.cc",
+    "$root_gen_dir/xwalk/resources/grit/xwalk_resources_map.h",
+  ]
+  deps = [
+    ":generate_xwalk_resources",
+  ]
+}
+
+grit("generate_xwalk_strings") {
```

Comment:
> I found the generate_xwalk_strings is much like targets in chrome/app/BUILD.gn.

First of all, did you understand the relationship between all the `*_strings` targets in `chrome/app/BUILD.gn` and the calls to `chrome_repack_locales` in `chrome/BUILD.gn`? Each `*_strings` target processes a different .grd file and produces several pak files out of it. The `chrome_repack_locales("repack_locales_pack")` call then goes through each locale and merges all those .pak files generated in the other targets into a single pak per locale.

> The problem here is what xwalk_win_zip.gypi(will be converted to gn format) wants in its "deps" list is a directory, but outputs of the grit or copy is a file-list. So I think the key is how to set the directory as an item of outputs in a target.
> I had a look on chrome_repack_locales, I understand how it defines its template, but I didn't find any similar thing helpful to our xwalk_string problem :(

`xwalk_win_zip.gypi` currently uses a directory as input because it's possible and easy with gyp. If it doesn't make sense with GN, we just use a list of files instead (remember what I said about not doing a word-by-word translation of the gyp files weeks ago?).

My whole point with my first comment in this line is that, according to what you are doing so far, there would be hardcoded lists of locales in at least 3 separate locations: `generate_xwalk_strings`, `copy_xwalk_strings` and the GN version of `xwalk_win_zip.gypi`. This is very error-prone and can be avoided if you keep one single list of locales somewhere (just like Chromium does with `build/config/locales.gni`) and use only that list all targets. And my suggestion to check `chrome_repack_locale` comes from the fact that that template takes care of setting the correct final output location of the pak files; even though we likely don't need to call grit's `repack.py` you can still take some ideas from that template code that can help you here.